freqtrade/freqai/utils.py

```diff
@@ -132,3 +137,58 @@ def get_required_data_timerange(
 #             trading_mode=config.get("trading_mode", "spot"),
 #             prepend=config.get("prepend_data", False),
 #         )
+
+
+def plot_feature_importance(model: Any, pair: str, dk: FreqaiDataKitchen,
+                            count_max: int = 25) -> None:
+    """
+        Plot Best and worst features by importance for a single sub-train.
+        :param model: Any = A model which was `fit` using a common library
+                            such as catboost or lightgbm
+        :param pair: str = pair e.g. BTC/USD
+        :param dk: FreqaiDataKitchen = non-persistent data container for current coin/loop
+        :param count_max: int = the amount of features to be loaded per column
+    """
+    from freqtrade.plot.plotting import go, make_subplots, store_plot_file
+
+    # Extract feature importance from model
+    models = {}
+    if 'FreqaiMultiOutputRegressor' in str(model.__class__):
+        for estimator, label in zip(model.estimators_, dk.label_list):
+            models[label] = estimator
+    else:
+        models[dk.label_list[0]]
+
+    for label in models:
+        mdl = models[label]
+        if "catboost.core" in str(mdl.__class__):
+            feature_importance = mdl.get_feature_importance()
+        elif "lightgbm.sklearn" or "xgb" in str(mdl.__class__):
+            feature_importance = mdl.feature_importances_
+        else:
+            logger.info('Model type not support for generating feature importances.')
+            return
+
+        # Data preparation
+        fi_df = pd.DataFrame({
+            "feature_names": np.array(dk.training_features_list),
+            "feature_importance": np.array(feature_importance)
+        })
+        fi_df_top = fi_df.nlargest(count_max, "feature_importance")[::-1]
+        fi_df_worst = fi_df.nsmallest(count_max, "feature_importance")[::-1]
+
+        # Plotting
+        def add_feature_trace(fig, fi_df, col):
+            return fig.add_trace(
+                go.Bar(
+                    x=fi_df["feature_importance"],
+                    y=fi_df["feature_names"],
+                    orientation='h', showlegend=False
+                ), row=1, col=col
+            )
+        fig = make_subplots(rows=1, cols=2, horizontal_spacing=0.5)
+        fig = add_feature_trace(fig, fi_df_top, 1)
+        fig = add_feature_trace(fig, fi_df_worst, 2)
+        fig.update_layout(title_text=f"Best and worst features by importance {pair}")
+
+        store_plot_file(fig, f"{dk.model_filename}-{label}.html", dk.data_path)
```

Comment:
i think this will store the result in the wrong location (depends where we expect it to end up at).

for model storage, we use `dk.data_path / f"{dk.model_filename}_svm_model.joblib"` (For example) - so they end up in a directory like `models/default/sub-train-CHZ_1663224320/...`.

while i can understand that we may not want them in the model dir itself (why not - the plot is JUST valid for this one model) - the current approach will store the model in "cwd" - which is wrong (we only store data in the `user_data_dir`).

The bigger problem is that "cwd" is not accessible on docker setups - so you're now generating plots in the container - which are never exposed (and can't be exposed as that location also contains a ton of other stuff) to the user system.

Reply:
Not really - it should be saving it to dk.data_path, which is with all the model specific files. 

I don't see where or why this is saving to any general directory.

Reply:
See the function arguments :)
https://github.com/freqtrade/freqtrade/blob/32c3f62934f97520490be1b009a5d0f5dc7dab4f/freqtrade/plot/plotting.py#L604